### ISSUE TITLE
interpolate 2x frame count output some black frames

### ISSUE THREAD

**Issue by abasgames** [nihui/dain-ncnn-vulkan#10]

**Comment by sbi6c** [comment]
the same by me on:

M:\vdo\+ dain-ncnn-vulkan-artifact-windows-2019>dain-ncnn-vulkan -i input_frames -o output_frames -v
[0 Intel(R) UHD Graphics 620]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[0 Intel(R) UHD Graphics 620]  bugsbn1=0  buglbia=0  bugcopc=0  bugihfa=0
[0 Intel(R) UHD Graphics 620]  fp16p=1  fp16s=1  fp16a=1  int8s=1  int8a=1
input_frames/frame_000001.png input_frames/frame_000002.png 0,000000 -> output_frames/000001.png done
input_frames/frame_000001.png input_frames/frame_000002.png 0,500000 -> output_frames/000002.png done
input_frames/frame_000002.png input_frames/frame_000003.png 0,000000 -> output_frames/000003.png done
input_frames/frame_000002.png input_frames/frame_000003.png 0,500000 -> output_frames/000004.png done
input_frames/frame_000003.png input_frames/frame_000004.png 0,000000 -> output_frames/000005.png done
input_frames/frame_000003.png input_frames/frame_000004.png 0,500000 -> output_frames/000006.png done
input_frames/frame_000004.png input_frames/frame_000005.png 0,000000 -> output_frames/000007.png done
![000001](https://user-images.githubusercontent.com/45274889/93153229-1c4c1e80-f701-11ea-8aff-4aeb48e3c21b.png)
![000002](https://user-images.githubusercontent.com/45274889/93153235-1eae7880-f701-11ea-9a15-43e4459baf4c.png)
![000003](https://user-images.githubusercontent.com/45274889/93153236-1f470f00-f701-11ea-9bda-577fa80302a5.png)
![000004](https://user-images.githubusercontent.com/45274889/93153240-20783c00-f701-11ea-8e9e-b03e6c247de0.png)
![000005](https://user-images.githubusercontent.com/45274889/93153241-21a96900-f701-11ea-9fd7-32db13a9a4df.png)
![000006](https://user-images.githubusercontent.com/45274889/93153243-22da9600-f701-11ea-8103-d2bf1260fba4.png)
![000007](https://user-images.githubusercontent.com/45274889/93153245-24a45980-f701-11ea-8901-1785e6df4854.png)

**Comment by YiWeiHuang-stack** [comment]
you need a NVIDIA card

**Comment by Keynsie** [comment]
> you need a NVIDIA card

The whole point of this is that it runs on Vulkan which doesn't need Nvidia cards

**Comment by Keynsie** [comment]
Have you tried changing around the tile size and image formats?

**Comment by YiWeiHuang-stack** [comment]
I tried to set tile size to 128,but output black frames

**Comment by YiWeiHuang-stack** [comment]
![000002](https://user-images.githubusercontent.com/68835291/93487332-bc8b8880-f937-11ea-86df-e52b1e958ab7.png)

**Comment by YiWeiHuang-stack** [comment]
I can use NVIDIA's graphics card to generate images normally, but it prompts vkwaitforfences failed and vkqueuesubmit failed, although I have set tile size to 128

**Comment by YiWeiHuang-stack** [comment]
maybe you can try the cain-ncnn-vulkan to instead dain-ncnn-vulkan

**Comment by sbi6c** [comment]
@YiWeiHuang-stack this is vulkan implementation, for compatibility see here [https://en.wikipedia.org/wiki/Vulkan_(API)#Hardware](url)

**Comment by lextra2** [comment]
I ran into the same issue. Make sure to close all applications that use 3D acceleration e.g. Discord, Chrome Browser, etc.

After doing that I no longer got black images. Also, like already said, reduce tile sizes to 128 if your GPU has low VRAM. Once the dain app runs out of memory you will get black images.

**Comment by sbi6c** [comment]
@lextra2 thx very much, I have tried both your suggestins. Few minor things changed, will post the screenshots. Ran it on Intel UHD 620 on 32GB RAM system, the task manager showed that gfx card took 1GB from it as shared, hmm.

**Comment by lextra2** [comment]
@sbi6c is your iGPU the only gpu in your system? I suspect some kind of intel driver bug here.

@nihui maybe you could take a look?

**Comment by sbi6c** [comment]
@lextra2 @nihui its the only one, yesterday I have updated to latest drivers ver., but with the same results, hmm, maybe something others local

**Comment by sbi6c** [comment]
@nihui @lextra2 guys, I have prepared the result samples how it ran w. t=128, and as much apps off as possible, incl. f.lux and Avira antivir, ThrottleStop, browsers. I have tried both DAIN windows2019 as well as windows 2016 version with same results (running on Win10 build 1903). The change is in 2 tiles computed, it seems more correct as the "black" images, but the colors are not correct. The non-black tiles are exactly 128x128 px (DAIN 't' parameter has been set to these 128)
